### DISCUSSION TITLE
LLM Chain + Chain Tool + Agent?

### DISCUSSION THREAD

**Discussion by rnk-ai** [FlowiseAI/Flowise#717] [Q&A]
I am trying to create an agent that has access to an LLM chain, connected by a Chain Tool. (ie. User asks to write something, then it goes through LLM chain or writer, then editor, and generates output).

But whenever I connect this to any agent, the response is chain.run is not a function.

Is there a way t o get this to work?

**Comment by HenryHengZJ** [comment]
must be some misconfiguration on the chatflow, if you can export and share it to see if we can replicate the issue

**Comment by b18x** [comment]
Having the same problem with the following flow, can you have a look? 

[Contextual Environment 2 Chatflow (3).json](https://github.com/FlowiseAI/Flowise/files/15077930/Contextual.Environment.2.Chatflow.3.json)
